lib/App/Critique/Command/prune.pm

```diff
@@ -48,34 +48,10 @@ sub execute {
     my $filter;
 
     if ( my $f = $opt->filter ) {
-        $filter =  sub {
-            my $path     = $_[0]->path->stringify;
-            my $is_match = $opt->invert ? $path !~ /$f/ : $path =~ /$f/;
-            if ( $opt->verbose ) {
-                if ( $is_match ) {
-                    info('Matched, keeping file (%s) ', $path);
-                }
-                else {
-                    info('Not matched, pruning file (%s) ', $path);
-                }
-            }
```

Comment:
This `verbose` conditional is also common between the two and can be generalized into a `success` and `failure` callback.